sdk/go/x/auto/stack.go

```diff
@@ -722,24 +799,26 @@ type PropertyDiff struct {
 
 // PreviewResult is the output of Stack.Preview() describing the expected set of changes from the next Stack.Up()
 type PreviewResult struct {
-	Steps         []PreviewStep  `json:"steps"`
-	ChangeSummary map[string]int `json:"changeSummary"`
+	StdOut        string
+	StdErr        string
+	ChangeSummary map[string]int
+	EventLog      []apitype.EngineEvent
```

Comment:
I think you probably have the tradeoff considered appropriately here but since you can pass event channels in the options anyway, do you think getting a full copy of the events in return as well seems a bit heavyweight?

Reply:
Yeah, this is the thing that I am most on the fence about. The [nodejs PR description](https://github.com/pulumi/pulumi/pull/6454#issue-583398443) best encapsulates my hesitation here. I agree that the full copy is overkill, what I really want is just the resource events and only really in the `Preview` case when there isn't an `UpdateSummary`. But it's probably "good enough" that you can get that from the event channel should you want it.

Reply:
Actually, thinking about it more I'm inclined to leave it in since we do include stdout and stderr on the result object (even though users could have the same access via `ProgressStreams`). Including `EventLog` on the `Result` object would be most consistent.

Reply:
Nope, changed my mind again. `EventLog` is too noisy, gonna exclude it from the `Result` and leave it as an opt-in via the channels.